packages/pages/src/upgrade/pagesUpdater.ts

```diff
@@ -43,46 +119,217 @@ const processDirectoryRecursively = (currentPath: string) => {
       );
     }
-    fs.writeFileSync(filePath, modifiedContent, "utf8"); // Update the file content
-    console.log(`Imports replaced in: ${filePath}`);
-  } catch (e) {
-    console.error("Error processing file: ", (e as Error).message);
+  };
+  processDirectoryRecursively(source, operation);
+};
+
+/**
+ * Replaces imports for pages/components with pages-components
+ * @param source the src folder
+ * @return hasReplaced
+ */
+export const replacePagesSlashComponentsImports = (source: string): boolean => {
+  let hasReplaced = false;
+  const operation = async (filePath: string) => {
+    const fileContent = fs.readFileSync(filePath, "utf8");
+    if (fileContent.match(pagesSlashComponentsRegex)) {
+      const modifiedContent = fileContent.replace(
+        pagesSlashComponentsRegex,
+        pagesComponentsReplacement
+      );
+      hasReplaced = true;
+      fs.writeFileSync(filePath, modifiedContent, "utf8");
+      console.log(`pages/components imports replaced in: ${filePath}`);
+    }
+  };
+  processDirectoryRecursively(source, operation);
+  return hasReplaced;
+};
+
+/**
+ * Replaces imports for sites-components with pages-components
+ * @param source the src folder
+ * @return hasReplaced
+ */
+export const replaceSitesComponentsImports = (source: string): boolean => {
+  let hasReplaced = false;
+  const operation = async (filePath: string) => {
+    const fileContent = fs.readFileSync(filePath, "utf8");
+    if (fileContent.match(sitesComponentsRegex)) {
+      const modifiedContent = fileContent.replace(
+        sitesComponentsRegex,
+        pagesComponentsReplacement
+      );
+      fs.writeFileSync(filePath, modifiedContent, "utf8");
+      hasReplaced = true;
+      console.log(`sites-components imports replaced in: ${filePath}`);
+    }
+  };
+  processDirectoryRecursively(source, operation);
+  return hasReplaced;
+};
+
+/**
+ * Replaces imports for react-components with pages-components
+ * @param source the src folder
+ * @return hasReplaced
+ */
+export const replaceReactComponentsImports = (source: string): boolean => {
+  let hasReplaced = false;
+  const operation = async (filePath: string) => {
+    const fileContent = fs.readFileSync(filePath, "utf8");
+    if (fileContent.match(reactComponentsRegex)) {
+      const modifiedContent = fileContent.replace(
+        reactComponentsRegex,
+        pagesComponentsReplacement
+      );
+      fs.writeFileSync(filePath, modifiedContent, "utf8");
+      hasReplaced = true;
+      console.log(`react-components imports replaced in: ${filePath}`);
+    }
+  };
+  processDirectoryRecursively(source, operation);
+  return hasReplaced;
+};
+
+/**
+ * Removes old fetch import that is no longer used
+ * @param root folder that contains package.json and tsconfig.json
+ */
+export const removeFetchImport = (root: string) => {
+  let hasRemoved = false;
+  const project = new Project({
+    tsConfigFilePath: path.resolve(root, "tsconfig.json"),
```

Comment:
Does this read the tsconfig to find the source files? Wouldn't we only want to do the source files defined by the projectStructure?

Reply:
Yes,

ts-morph wants a ts-config, but we can also set a bunch of constants instead

This gives us all of the source files we are looking for however

Reply:
Updated to use project structure instead